File: rest/src/main/java/com/grupo3/meetings/repository/UserRepository.java
package com.grupo3.meetings.repository;

import com.grupo3.meetings.domain.Event;
import com.grupo3.meetings.domain.User;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

@Repository
public class UserRepository {
        List<User> db ;
    public Optional<User> findUserById(String userId) {
        return Optional.empty();
    }
    public UserRepository() {
        this.db = new ArrayList<User>();
    }
}

File: rest/src/main/java/com/grupo3/meetings/repository/UserRepository.kt
package com.grupo3.meetings.repository

import com.grupo3.meetings.domain.dto.UserDTO
import com.grupo3.meetings.domain.models.User
import com.grupo3.meetings.domain.models.UserId

interface UserRepository {

  fun createUser(user: UserDTO): User
  fun updateUser(user: User): User?
  fun getByEmail(email: String): User?
  fun getById(id: UserId): User?
  fun getAll(): List<User>
}

var userRepository = InMemoryUserRepository()
